### ISSUE TITLE
Negative Items in Compacting Drawers

### ISSUE THREAD

**Issue by Seyeght** [jaquadro/StorageDrawers#465]
Slime blocks when in a Compacting drawer will stay positive until it gets above 127 blocks, then will go negative and count back up to zero. 
![negativedrawer](https://cloud.githubusercontent.com/assets/20807322/24843742/382c7c12-1d6b-11e7-965b-d5cbd8d19599.PNG)

I loaded it with Blue Slime balls, which have a 3x3 pattern into Minecraft Green slime blocks, but after 127 blocks, on the 128th it switches over to -128 and swaps the image to the Minecraft green slime ball. When withdrawing from the drawer it still pulls out the blue slime.

This was working as intended in v.1.10.2-3.6.3, but isnt working in v.1.10.2-3.7.0

**Comment by Seyeght** [comment]
I think part of the problem is that there are 2 ways to make the Minecraft Slime block, one that is 2x2 of Minecraft Green Slime Balls, and one that is 3x3 of Tinkers Slime Balls.

**Comment by Seyeght** [comment]
In previous versions it was like the Compacting drawer was based on the bottom left item, but in this version it seems to be doing the calculations off the top item. which because of the dual recipes is causing the visual glitch for the slime ball and the negative item counter.

**Comment by jaquadro** [comment]
Can you post log output after enabling debug logging in the mod options?  That will give some insight into what it's doing with recipes (I don't have TE in my dev environment right now, and it's behaving correctly with just vanilla slime stuff)

**Comment by zISKo666** [comment]
I have the same problem not only with compacting drawers.
I tried it with all types of drawers and they behave all the same way.

Using mod version 1.10.2-3.7.1

enabling debug mode just gives:
[09:46:57] [Client thread/INFO] [StorageDrawers/]: BlockDrawers.onBlockActivated
[09:46:57] [Client thread/INFO] [StorageDrawers/]:   null item

I'll try to explain with screenshots:

top left corner (The One Probe HUD) shows 3562 items
and the level indicator show the correct fill level
![2017-04-14_10 21 35](https://cloud.githubusercontent.com/assets/6992216/25037961/59c9a482-20fc-11e7-8af0-f67be3980924.png)

shift+right clicking shows -22 items
![2017-04-14_10 21 40](https://cloud.githubusercontent.com/assets/6992216/25037970/64c3e762-20fc-11e7-9ce3-9b0ed943abf0.png)

now the level indicator shows an empty drawer
but the top left HUD shows the correct amount of items
![2017-04-14_10 21 43](https://cloud.githubusercontent.com/assets/6992216/25038079/09c1354e-20fd-11e7-8ea9-0851e77a5e18.png)

removing and replacing and upgrade e.g. the the status upgrade ...
![2017-04-14_10 21 52](https://cloud.githubusercontent.com/assets/6992216/25038115/4062eb6a-20fd-11e7-9d7f-3ff36d552634.png)
![2017-04-14_10 21 58](https://cloud.githubusercontent.com/assets/6992216/25038120/454ecaea-20fd-11e7-9d7f-5c6ecf674733.png)

... turns everything back to normal
![2017-04-14_10 22 02](https://cloud.githubusercontent.com/assets/6992216/25038172/8d95a3d2-20fd-11e7-8506-47225814def7.png)

**Comment by jaquadro** [comment]
When you say it turns back to normal, so you just mean the fill bar, or does it fix the number readout in the interface?

**Comment by zISKo666** [comment]
No, just the fill bar.
When you enter the interface you can read the correct numbers for a tens of a second then they change to the false ones.

**Comment by jaquadro** [comment]
So far I haven't been able to reproduce this.  Are either of you able to reproduce this with just SD installed?

**Comment by JaSpr** [comment]
For me, this issue ONLY reproduces when I shift-right-click the drawer.  Before doing that, the count is correct.  Changing the upgrades almost always resolves the count, and the attached Refined Storage system *always* has the correct count.

Pack: FTB present Sky Factory 3 (3.0.9)
Storage Drawers: 1.10.2-3.7.1
Forge: 12.18.3.2281

Server:
    SpongeAPI: 5.2.0-SNAPSHOT-a3257a0
    SpongeForge: 1.10.2-2281-5.2.0-BETA-2274
    Minecraft Forge: 12.18.3.2254

**Comment by JaSpr** [comment]
Also, it's not just negative numbers, but also very low numbers.  I have one that has like 2k of stuff (compacted) and when I right click, it drops to 4

**Comment by jaquadro** [comment]
I got far enough to see a variation of this (not seeing negative numbers print, but seeing them drop to 0 and negative numbers flying around behind the scenes).

The cause is using Mojang's built-in inventory window packets, which send stack size data as a byte.  I can't reproduce it reliably with all items (e.g. I get the zeroing behavior with slime and quartz, but not gold.  Pretty odd).

I think I ran into this problem some place else ... or maybe that's part of the code that got nuked with the 3.7.0 update.
